eclair-rpc/src/main/scala/org/bitcoins/eclair/rpc/json/EclairModels.scala

```diff
@@ -246,10 +237,22 @@ case class ChannelResult(
     state: ChannelState,
     feeBaseMsat: Option[MilliSatoshis],
     feeProportionalMillionths: Option[FeeProportionalMillionths],
-    data: JsObject)
+    data: JsObject) {
+  import JsonReaders._
+  def shortChannelId: ShortChannelId = (data \ "shortChannelId").validate[ShortChannelId].get
+}
 
 // ChannelResult ends here
 
+case class CreateInvoiceResult(
+  prefix: String,
+  timestamp: Long,
+  nodeId: NodeId,
+  serialized: String,
+  description: String,
+  paymentHash: Sha256Digest,
+  expiry: Long)
```

Comment:
better type?

Reply:
Again, which one? It's always easy to create any type of date/time object from a Long.

Reply:
So the interesting thing about this field is i believe it is in seconds not milliseconds.

https://github.com/lightningnetwork/lightning-rfc/blob/master/11-payment-encoding.md#tagged-fields

So it is sort of a foot gun for people using the api if we just pass around `Long` instead of something more structured that captures the fact that this field is in seconds. My go to for things like this is `FiniteDuration` but can be convinced otherwise